### PR TITLE
fix(extra artifacts): avoid inappropriate builds

### LINKED ISSUE #658: `cargo dist build --artifacts=local` builds extra artifacts?
It seems like `cargo dist build --artifacts=local` builds artifacts specified in [`extra-artifacts`](https://opensource.axo.dev/cargo-dist/book/reference/config.html#extra-artifacts) in addition to local artifacts--is this intentional? My impression was that `--artifacts=local` would not build `extra-artifacts`, while `--artifacts=global` would.

### PR DESCRIPTION
**Pull request by mistydemeo**
We had an issue where we would still build extra artifacts even if they'd been filtered out of the target artifact list. This fixes that.

The issue was caused by the separate logic we had to use to add these builds to the graph. Because each of these builds produces multiple artifacts, we weren't able to add them in the same place that other global artifacts are added. However, that meant we weren't taking into account the list of target artifacts - we were unconditionally building them, even if we didn't want them.

Also fixes an issue where they were assigned to the local build array instead of the global build array, though the impact of that bug was minor.

Fixes #658.

### CODE REVIEW

File: cargo-dist/src/tasks.rs [hunk 1]
```diff
@@ -2274,7 +2284,7 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
             axoproject::WorkspaceKind::Rust => self.compute_cargo_builds(),
         };
         local_build_steps.extend(builds);
-        local_build_steps.extend(self.compute_extra_builds());
+        global_build_steps.extend(self.compute_extra_builds());
```
Comment (Gankra) [inline cargo-dist/src/tasks.rs:2287]: ...right that makes sense

**Review by Gankra — APPROVED** [review]
tentatively approved but that string thing is uhhh

File: cargo-dist/src/tasks.rs [hunk 2]
```diff
@@ -2252,9 +2252,19 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
     }
 
     fn compute_extra_builds(&mut self) -> Vec<BuildStep> {
+        let artifacts = self
+            .inner
+            .artifacts
+            .iter()
+            .map(|artifact| artifact.id.clone())
+            .collect::<String>();
```
[the remark this answers is not in the document]
Comment (max-heller) [reply cargo-dist/src/tasks.rs:2260]: how about collecting into a `HashSet<&str>`?

[the remark this answers is not in the document]
Comment (mistydemeo) [reply cargo-dist/src/tasks.rs:2260]: Oh yeah, this is a typo. I meant to collect this into an array of strings, not a single string. Will fix.

**Pull request merged**